src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs

```diff
@@ -1007,6 +1093,186 @@ static void M()
                 Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "object? x").WithArguments("System.Runtime.CompilerServices.NullableAttribute").WithLocation(5, 16));
         }
 
+        [Fact]
+        public void Tuples()
+        {
+            var source =
+@"public class A
+{
+    public static ((object?, object) _1, object? _2, object _3, ((object?, object), object?) _4) Nested;
+    public static (object? _1, object _2, object? _3, object _4, object? _5, object _6, object? _7, object _8, object? _9) Long;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
+            CompileAndVerify(comp, validator: assembly =>
+            {
+                var reader = assembly.GetMetadataReader();
+                var typeDef = GetTypeDefinitionByName(reader, "A");
+                var fieldDefs = typeDef.GetFields().Select(f => reader.GetFieldDefinition(f)).ToArray();
+
+                // Nested tuple
+                var field = fieldDefs.Single(f => reader.StringComparer.Equals(f.Name, "Nested"));
+                var customAttributes = field.GetCustomAttributes();
+                AssertAttributes(reader, customAttributes,
+                    "MemberReference:Void System.Runtime.CompilerServices.TupleElementNamesAttribute..ctor(String[])",
+                    "MethodDefinition:Void System.Runtime.CompilerServices.NullableAttribute..ctor(Boolean[])");
+                var customAttribute = reader.GetCustomAttribute(customAttributes.ElementAt(1));
+                AssertEx.Equal(ImmutableArray.Create(false, false, true, false, true, false, false, false, true, false, true), reader.ReadBoolArray(customAttribute.Value));
+
+                // Long tuple
+                field = fieldDefs.Single(f => reader.StringComparer.Equals(f.Name, "Long"));
+                customAttributes = field.GetCustomAttributes();
+                AssertAttributes(reader, customAttributes,
+                    "MemberReference:Void System.Runtime.CompilerServices.TupleElementNamesAttribute..ctor(String[])",
+                    "MethodDefinition:Void System.Runtime.CompilerServices.NullableAttribute..ctor(Boolean[])");
+                customAttribute = reader.GetCustomAttribute(customAttributes.ElementAt(1)); 
+                AssertEx.Equal(ImmutableArray.Create(false, true, false, true, false, true, false, true, false, false, true), reader.ReadBoolArray(customAttribute.Value));
+            });
+            var source2 =
+@"class B
+{
+    static void Main()
+    {
+        A.Nested._1.Item1.ToString(); // 1
```

Comment:
Nit: thanks for the labels/comments. One suggestion to make them even more usable next time: either don't mark the lines that produce no warnings, or mark them `// ok`. That allows less mental tracking for reviewers.